hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/FinalizeStatusSubCommand.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.ozone.admin.om;
+
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.ozone.om.protocol.OzoneManagerProtocol;
+import org.apache.hadoop.ozone.upgrade.UpgradeFinalizer;
+import picocli.CommandLine;
+
+import java.util.UUID;
+import java.util.concurrent.Callable;
+
+/**
+ * Handler of ozone admin om finalizestatus command.
+ */
+@CommandLine.Command(
+    name = "finalizestatus",
+    description = "Get the finalize status of om cluster.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+public class FinalizeStatusSubCommand implements Callable<Void> {
+
+  @CommandLine.ParentCommand
+  private OMAdmin parent;
+
+  @CommandLine.Option(
+      names = {"-id", "--service-id"},
+      description = "Ozone Manager Service ID"
+  )
+  private String omServiceId;
+
+  @CommandLine.Option(
+      names = {"-host", "--service-host"},
+      description = "Ozone Manager Host"
+  )
+  private String omHost;
+
+  @Override
+  public Void call() throws Exception {
+    OzoneManagerProtocol client =
+        parent.createOmClient(omServiceId, omHost, false);
+    String upgradeClientID = "Upgrade-Client-" + UUID.randomUUID().toString();
+    UpgradeFinalizer.StatusAndMessages progress =
+        client.queryUpgradeFinalizationProgress(upgradeClientID, true);
```

Comment:
Can we not "takeover" the finalization from another client, just to get status? If there is a finalization going on currently, then that will fail since the upgradeClientId has changed. Maybe we could pass in a readOnly flag, and use that in the server to just query status instead of taking over. What do you think?

Reply:
@avijayanhwx oh, got it, I shall add some logic to diff a takeover and a read-only status, thx~

Reply:
https://github.com/apache/ozone/blob/ed00db7a52bf1acce9a82b21642032b42c0fb223/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/upgrade/UpgradeFinalizer.java#L169

As I read through the comments of the interface, I found that the assert on clientId might be optional by design?
Suppose we could just remove the assertion in `BasicUPgradeFinalizer` and let any client fetch the status of finalization? I'm not sure whether it will cause any problems?

Reply:
@guihecheng It is optional at the interface level since we could have implementors (maybe recon or test finalizers that don't need it) I believe the takeover flag must remain since it used on the update path. The readOnly flag could just skip the assert on clientId. What do you think?

cc @fapifta who implemented this initially.

Reply:
Hi @avijayanhwx , updated with a readonly flag, skip on assertion when readonly is true, thx~